test/Util.test.js

```diff
@@ -0,0 +1,241 @@
+      bar.set('a', 1);
+      bar.set('b', 2);
+      bar.set('c', 3);
+
+      const obj = {
+        foo: 'foo',
+        bar,
+      };
+
+      expect(Util.flatten(obj)).toStrictEqual({ foo: 'foo', bar: ['a', 'b', 'c'] });
+    });
+
+    test('only include specific props', () => {
+      const obj = {
+        foo: 'foo',
+        bar: 'bar',
+        baz: 'baz',
+      };
+
+      expect(Util.flatten(obj, { foo: true, bar: false, baz: false })).toStrictEqual({ foo: 'foo' });
+    });
+
+    test('ignore props that start with an underscore', () => {
+      const bar = new Collection();
+      bar.set('a', 1);
+      bar.set('b', 2);
+      bar.set('c', 3);
+
+      const obj = {
+        foo: 'foo',
+        bar,
+        _ignore: 'me',
+      };
+
+      expect(Util.flatten(obj)).toStrictEqual({ foo: 'foo', bar: ['a', 'b', 'c'] });
+    });
+  });
+
+  test('Util#splitMessage', () => {
+    const text = `${'x'.repeat(1955)}\n${'x'.repeat(1984)}`;
+    const split = Util.splitMessage(text);
+    expect(split.length).toStrictEqual(2);
+    expect(split[0].length).toStrictEqual(1955);
+    expect(split[1].length).toStrictEqual(1984);
+  });
+
+  describe('Util#escapeMarkdown', () => {
+    test('escape a code block', () => {
+      const input = '```\nhello```';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\`\\`\\`\nhello\\`\\`\\`');
+    });
+
+    test('escape code block content', () => {
+      const input = '```\n**hello**```';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\`\\`\\`\n\\*\\*hello\\*\\*\\`\\`\\`');
+    });
+
+    test('escape an inline code block', () => {
+      const input = '`hello`';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\`hello\\`');
+    });
+
+    test('escape inline code block content', () => {
+      const input = '`**hello**`';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\`\\*\\*hello\\*\\*\\`');
+    });
+
+    test('escape bold text', () => {
+      const input = '**hello**';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\*\\*hello\\*\\*');
+    });
+
+    test('escape italic text (underscores)', () => {
+      const input = '_hello_';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\_hello\\_');
+    });
+
+    test('escape italic text (asterisks)', () => {
+      const input = '*hello*';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\*hello\\*');
+    });
+
+    test('escape underlined text', () => {
+      const input = '__hello__';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\_\\_hello\\_\\_');
+    });
+
+    test('escape strikethrough text', () => {
+      const input = '~~hello~~';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\~\\~hello\\~\\~');
+    });
+
+    test('escape spoiled text', () => {
+      const input = '||hello||';
+      expect(Util.escapeMarkdown(input)).toStrictEqual('\\|\\|hello\\|\\|');
+    });
+  });
+
+  describe('Util#parseEmoji', () => {
```

Comment:
When would an emoji ever be URI encoded?
https://github.com/discordjs/discord.js/blob/1be67b88516b104073e46574a180498bad2aa02b/src/util/Util.js#L262-L263